api-gateway/src/lib.rs

```diff
@@ -193,3 +197,21 @@ fn get_response_headers() -> HashMap<String, Vec<String>> {
     );
     headers
 }
+
+async fn topup_balance(_ctx: &Context, _request: DepositRequest) -> RpcResult<HttpResponse> {
+    let balance = BalanceManagerSender::to_actor("balancemanager")
+        .deposit(_ctx, &_request)
+        .await?;
+
+    let mut headers: HashMap<String, Vec<String>> = HashMap::new();
```

Comment:
please use get_response_headers() inside json_with_headers param directly, this code is extracted to it already